### PR TITLE
Create a util for deploying or committing to models

### PR DESCRIPTION
**Pull request by huwshimi**
Move the deploy/commit functionality from the old deployment flow (https://github.com/juju/juju-gui/blob/develop/jujugui/static/gui/src/app/components/deployment/summary/summary.js#L199) to a util so that it can be used in the new flow.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA: none as this util is not being used yet.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5513/
Test PASSed.

File: jujugui/static/gui/src/app/views/utils.js [hunk 1]
```diff
@@ -2252,6 +2252,102 @@ YUI.add('juju-view-utils', function(Y) {
...
+      callback();
+    });
+  };
+
+  /**
+    Detach the handler for committing the changeset on login.
+
+    @method _detachOnLoginhandler
+  */
+  utils._detachOnLoginhandler = function() {
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/views/utils.js]: detachOnLoginHandler? Capital H

File: jujugui/static/gui/src/app/views/utils.js [hunk 2]
```diff
@@ -2252,6 +2252,102 @@ YUI.add('juju-view-utils', function(Y) {
...
+    @param {Function} callback The function to be called once the deploy is
+      complete.
+    @param {String} model The name of the new model.
+    @param {String} credential The credentials to be used with the new model.
+    @param {String} cloud The cloud to deploy to.
+    @param {String} region The cloud region to deploy to.
+  */
+  utils.deploy = function(
+    env, jem, users, autoPlaceUnits, createSocketURL, appSet, committed,
+    callback, autoplace=true, model, credential, cloud, region) {
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/views/utils.js]: Missing docs for the autoplace param. What is the difference between autoplace and autoPlaceUnits?

File: jujugui/static/gui/src/app/views/utils.js [hunk 3]
```diff
@@ -2252,6 +2252,102 @@ YUI.add('juju-view-utils', function(Y) {
...
+    @param {Function} createSocketURL The method used to create a socket URL.
+    @param {Function} appSet The method used to set parameters on the app.
+    @param {Function} callback The function to be called once the deploy is
+      complete.
+    @param {Object} error The model creation error.
+    @param {Object} model The newly created model data.
+  */
+  utils._newModelCallback = function(
+    env, createSocketURL, appSet, callback, error, model) {
+    if (error) throw error;
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/views/utils.js]: Would this result in a user notification?

File: jujugui/static/gui/src/app/views/utils.js [hunk 4]
```diff
@@ -2252,6 +2252,102 @@ YUI.add('juju-view-utils', function(Y) {
...
+    var pathParts = model.hostPorts[0].split(':');
+    // Set the credentials in the env so that the GUI
+    // is able to connect to the new model.
+    env.setCredentials({
+      user: 'user-' + model.user,
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/views/utils.js]: Out of curiosity. Why do we prefix the username with `user-`?

**Comment by anthonydillon** [comment]
Thanks @huwshimi, few questions in the code but apart from that :+1:

File: jujugui/static/gui/src/app/views/utils.js (hunk 2 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/views/utils.js]: Ah oops, `autoplace` is a flag for whether we should place unplaced units, `autoPlaceUnits` is the method for doing so.

File: jujugui/static/gui/src/app/views/utils.js (hunk 3 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/views/utils.js]: Not automatically, but it does mean that if when this util is used it is wrapped in a try/catch block the catch can output a message to the user.

File: jujugui/static/gui/src/app/views/utils.js (hunk 4 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/views/utils.js]: I think juju's internal format has the usernames prefixed with user-, not sure why...

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5520/
Test PASSed.

**Comment by hatched** [comment]
Thanks for putting this together @huwshimi :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**